.github/workflows/release.yml

```diff
@@ -0,0 +1,54 @@
+name: Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      tag:
+        type: string
+        required: true
+  release:
+    types: [published]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ inputs.tag || github.event.release.tag_name }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    name: Build GDExtension
+    uses: ./.github/workflows/build.yml
+
+  archive:
+    needs: build
+    name: Create & Upload Archive
+    runs-on: ubuntu-latest
+    env:
+      RELEASE_TAG: ${{ inputs.tag || github.event.release.tag_name }}
+      ARTIFACT_NAME: ${{ needs.build.outputs.artifact-name }}
+      ADDON_PATH: addons/${{ needs.build.outputs.artifact-name }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ env.RELEASE_TAG }}
+          # We only need the addon folder in the release archive.
+          # This minimises what's checked out.
+          sparse-checkout: demo/${{ env.ADDON_PATH }}
+      - name: Download Binaries
+        uses: actions/download-artifact@v4
+        with:
+          pattern: ${{ env.ARTIFACT_NAME }}-*
+          path: demo/${{ env.ADDON_PATH }}/lib
+          merge-multiple: true
+      - id: create-archive
+        name: Create Archive
+        shell: sh
+        working-directory: ./demo
+        env:
+          ARCHIVE_NAME: ${{ env.ARTIFACT_NAME }}-${{ env.RELEASE_TAG }}.tar.gz
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          echo "Creating $ARCHIVE_NAME from ${{ env.ADDON_PATH }}"
+          tar -czvf $ARCHIVE_NAME ${{ env.ADDON_PATH }}
```

Comment:
We probably need to switch this to a zip file, as that's what every addon seems to use.